libsolidity/NameAndTypeResolver.cpp

```diff
@@ -108,13 +108,11 @@ void NameAndTypeResolver::importInheritedScope(ContractDefinition const& _base)
 	auto iterator = m_scopes.find(&_base);
 	solAssert(iterator != end(m_scopes), "");
 	for (auto const& nameAndDeclaration: iterator->second.getDeclarations())
-	{
-		Declaration const* declaration = nameAndDeclaration.second;
-		// Import if it was declared in the base, is not the constructor and is visible in derived classes
-		if (declaration->getScope() == &_base && declaration->getName() != _base.getName() &&
-				declaration->isVisibleInDerivedContracts())
-			m_currentScope->registerDeclaration(*declaration);
-	}
+		for (auto const& declaration: nameAndDeclaration.second)
+			// Import if it was declared in the base, is not the constructor and is visible in derived classes
+			if (declaration->getScope() == &_base && declaration->getName() != _base.getName() &&
```

Comment:
The name check can now be removed (it is already done in `isVisibleInDeriedContracts`).